tests/plugins/find_and_replace.test.ts

```diff
@@ -145,133 +218,200 @@ describe("basic search", () => {
 
   test("search on empty string clears matches", () => {
     model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
-    expect(model.getters.getSearchMatches()).toHaveLength(4);
+    expect(model.getters.getSearchMatches()).toHaveLength(3);
     model.dispatch("UPDATE_SEARCH", { toSearch: "", searchOptions });
     expect(model.getters.getSearchMatches()).toHaveLength(0);
   });
 
-  test.skip("Will search a modified cell", () => {
-    // not implemented
+  test("start searching from current sheet and update index", () => {
+    searchOptions.searchScope = "allSheets";
+    activateSheet(model, "s2");
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    const matches = model.getters.getSearchMatches();
+    const matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(4);
+    expect(matchIndex).toStrictEqual(3);
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: false });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+    expect(matches[3]).toStrictEqual({ sheetId: "s2", col: 0, row: 1, selected: true });
+  });
+
+  test("modifying cells update the search", () => {
     model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
     let matches = model.getters.getSearchMatches();
     let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches.length).toBe(3);
+    expect(matchIndex).toStrictEqual(0);
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+    setCellContent(model, "A2", "hello");
+    setCellContent(model, "B1", "1");
+    matches = model.getters.getSearchMatches();
+    matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches.length).toBe(3);
+    expect(matchIndex).toStrictEqual(0);
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 1, row: 0, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+  });
+
+  test("search begins from current sheet with index starting at the first sheet in allSheets search scope", () => {
```

Comment:
that's the exact same test as "start searching from current sheet and update index"?? with one more line ?